src/components/Table/useColumns.ts

```diff
@@ -0,0 +1,21 @@
+import { Dispatch, SetStateAction, useMemo, useState } from "react";
+import { GridColumn, Kinded } from "./GridTable";
+
+export interface Columns<R extends Kinded, S = {}> {
+  allColumns: GridColumn<R, S>[];
+  visibleColumns: GridColumn<R, S>[];
+}
+
+export function useColumns<R extends Kinded, S = {}>(
+  tableColumns: GridColumn<R, S>[],
+): [Columns<R, S>, Dispatch<SetStateAction<Columns<R, S>>>] {
+  const initColumns: Columns<R, S> = useMemo(() => {
+    return {
+      allColumns: tableColumns,
+      visibleColumns: tableColumns.filter((column) => (column.canHide && column.visible) || !column.canHide),
```

Comment:
**suggestion:** Same we earlier, I think we can make this a bit easier to read via:
```suggestion
      visibleColumns: tableColumns.filter((column) => column.canHide ? column.visible : true,
```